Build failure: cutter

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *cutter*

### Build log

```
...
-- Found PkgConfig: /nix/store/39nq7p6jcm9m3xqqm99skwmgsqw4pbbg-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2")
-- Checking for module 'libgvc'
--   Found libgvc, version 10.0.1
-- Found Graphviz: /nix/store/fv0iivxdq5zl8dzkmgg16d8ldqqig7ln-graphviz-10.0.1/include/graphviz
-- 
-- Building Cutter version 2.3.4
-- Options:
-- - Bundled rizin: OFF
-- - Python: ON
-- - Python Bindings: ON
-- - KSyntaxHighlighting: OFF
-- - Graphviz: TRUE
-- - Downloads dependencies: OFF
-- - Enable Packaging: OFF
-- - Package Dependencies: OFF
-- - Package RzGhidra: OFF
-- - Package RzLibSwift: OFF
-- - Package RzLibYara: OFF
-- - Package RzSilhouette: OFF
-- - Package JSDec: OFF
-- - QT6: ON
-- 
-- Configuring done (8.8s)
-- Generating done (0.3s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
build flags: -j2 SHELL=/nix/store/dy2z01kpnxn7dn2kgfdxs4fm8xy9mb89-bash-5.2p26/bin/bash
[  0%] Built target Cutter_autogen_timestamp_deps
...
[ 69%] Building CXX object src/CMakeFiles/Cutter.dir/widgets/SdbWidget.cpp.o
[ 69%] Building CXX object src/CMakeFiles/Cutter.dir/plugins/PluginManager.cpp.o
[ 70%] Building CXX object src/CMakeFiles/Cutter.dir/common/BasicBlockHighlighter.cpp.o
/build/source/src/plugins/PluginManager.cpp: In member function 'CutterPlugin* PluginManager::loadPythonPlugin(const char*)':
/build/source/src/plugins/PluginManager.cpp:218:44: warning: 'using SbkObjectType = PyTypeObject' is deprecated [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations8;;]
  218 |             reinterpret_cast<SbkObjectType *>(SbkCutterBindingsTypes[SBK_CUTTERPLUGIN_IDX]),
      |                                            ^
In file included from /build/source/build/src/bindings/CutterBindings/cutterbindings_python.h:9,
                 from /build/source/src/plugins/PluginManager.cpp:6:
/nix/store/whispjhna99wpqqbrdllngj3a9akc22v-shiboken6-6.7.0/include/shiboken6/sbkpython.h:73:7: note: declared here
   73 | using SbkObjectType [[deprecated]] = PyTypeObject; // FIXME PYSIDE 7 remove
      |       ^~~~~~~~~~~~~
/build/source/src/plugins/PluginManager.cpp:218:47: error: 'SbkCutterBindingsTypes' was not declared in this scope; did you mean 'SbkCutterBindingsTypeStructs'?
  218 |             reinterpret_cast<SbkObjectType *>(SbkCutterBindingsTypes[SBK_CUTTERPLUGIN_IDX]),
      |                                               ^~~~~~~~~~~~~~~~~~~~~~
      |                                               SbkCutterBindingsTypeStructs
make[2]: *** [src/CMakeFiles/Cutter.dir/build.make:1924: src/CMakeFiles/Cutter.dir/plugins/PluginManager.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:103: src/CMakeFiles/Cutter.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
error: builder for '/nix/store/lfpqcqagpymsqlfh6cjlyql3fnkh45b9-cutter-2.3.4.drv' failed with exit code 2
error: 1 dependencies of derivation '/nix/store/vadfy934fqjvfvjx84bh39pxnjsa1lz4-cutter-2.3.4_fish-completions.drv' failed to build
error: 1 dependencies of derivation '/nix/store/w993xs3k7w6c9lp37i42hq2v0xxywwg7-man-paths.drv' failed to build
error: 1 dependencies of derivation '/nix/store/b71wla052qxivnwcpw0j9qblbncpj0qy-rz-ghidra-0.7.0.drv' failed to build
```

### Additional context

Add any other context about the problem here.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@dtzWill @Mic92 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.28, NixOS, 24.05 (Uakari), 24.05pre617818.58a1abdbae32`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"home-manager, nixos, nixpkgs"`
 - nixpkgs: `/home/athena/.nix-defexpr/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/shiboken6/default.nix
+++ b/pkgs/development/python-modules/shiboken6/default.nix
@@ -1,6 +1,7 @@
 {
   lib,
   fetchurl,
+  fetchpatch,
   llvmPackages,
   python,
   cmake,
@@ -23,7 +24,16 @@ stdenv'.mkDerivation rec {
 
   sourceRoot = "pyside-setup-everywhere-src-${version}/sources/${pname}";
 
-  patches = [ ./fix-include-qt-headers.patch ];
+  patches = [
+    ./fix-include-qt-headers.patch
+    # Remove this patch when updating to 6.8.0
+    (fetchpatch {
+      name = "backwards-compatibility-with-6.6.x.patch";
+      url = "https://code.qt.io/cgit/pyside/pyside-setup.git/patch/?id=4f9a20e3635f4f0957e0774588b1d9156e88a572";
+      hash = "sha256-B2jhLWopgaSF/rUXMZFPZArDUNojlBgn7kdVyQull+I=";
+      stripLen = 2;
+    })
+  ];
 
   nativeBuildInputs = [
     cmake

